tests/sentry/utils/performance_issues/test_slow_span_detector.py

```diff
@@ -0,0 +1,105 @@
+import unittest
+from typing import List
+
+import pytest
+
+from sentry.eventstore.models import Event
+from sentry.testutils.performance_issues.event_generators import EVENTS, create_event, create_span
+from sentry.testutils.silo import region_silo_test
+from sentry.utils.performance_issues.performance_detection import (
+    DetectorType,
+    GroupType,
+    PerformanceProblem,
+    SlowSpanDetector,
+    get_detection_settings,
+    prepare_problem_for_grouping,
+    run_detector_on_data,
+)
+
+
+@region_silo_test
+@pytest.mark.django_db
+class SlowSpanDetectorTest(unittest.TestCase):
+    def setUp(self):
+        super().setUp()
+        self.settings = get_detection_settings()
+
+    def find_slow_span_problems(self, event: Event) -> List[PerformanceProblem]:
+        detector = SlowSpanDetector(self.settings, event)
+        run_detector_on_data(detector, event)
+        # The usage of "prepare_problem_for_grouping" is only needed because the slow db detector is using PerformanceSpanProblem,
+        # TODO - refactor slow span detector to use PerformanceProblem
+        return [
+            prepare_problem_for_grouping(problem, event, DetectorType.SLOW_SPAN)
+            for problem in detector.stored_problems.values()
+        ]
```

Comment:
Slight diff in this detector vs others when doing this refactor, fyi @0Calories might be relevant to you!